### ISSUE TITLE
Dockerfile libsoup-ERROR **: 03:12:25.949: libsoup2 symbols detected. Using libsoup2 and libsoup3 in the same process is not supported.

### ISSUE THREAD

**Issue by mpisat** [meetecho/simple-whip-client#13]
I have the same issue in whep client. 

I built it with
docker build -t simple-whip-client .

frhb64652ds simple-whip-client # docker run -it --rm -e "URL=https://fr.xxx.com/api/whip?stream=ff_mur_test?token=9a58a1a2-f953-4904-97a6-35b3354c8ae2" simple-whip-client

--------------------
Simple WHIP client
------------------

WHIP endpoint:  https://fr.xxx.com/api/whip?stream=ff_mur_test?token=9a58a1a2-f953-4904-97a6-35b3354c8ae2
Bearer Token:   (none)
Trickle ICE:    yes (HTTP PATCH)
Auto STUN/TURN: no
STUN server:    stun://stun.l.google.com:19302
TURN server:    (none)
Audio pipeline: audiotestsrc is-live=true wave=red-noise ! audioconvert ! audioresample ! queue ! opusenc ! rtpopuspay pt=100 ssrc=1 ! queue ! application/x-rtp,media=audio,encoding-name=OPUS,payload=100
Video pipeline: videotestsrc is-live=true pattern=ball ! videoconvert ! queue ! vp8enc deadline=1 ! rtpvp8pay pt=96 ssrc=2 ! queue ! application/x-rtp,media=video,encoding-name=VP8,payload=96

[WHIP] Initializing the GStreamer pipeline:
webrtcbin name=sendonly bundle-policy=3  stun-server=stun://stun.l.google.com:19302  videotestsrc is-live=true pattern=ball ! videoconvert ! queue ! vp8enc deadline=1 ! rtpvp8pay pt=96 ssrc=2 ! queue ! application/x-rtp,media=video,encoding-name=VP8,payload=96 ! sendonly. audiotestsrc is-live=true wave=red-noise ! audioconvert ! audioresample ! queue ! opusenc ! rtpopuspay pt=100 ssrc=1 ! queue ! application/x-rtp,media=audio,encoding-name=OPUS,payload=100 ! sendonly.

(whip-client:7): libsoup-ERROR **: 03:14:03.702: libsoup2 symbols detected. Using libsoup2 and libsoup3 in the same process is not supported.
Trace/breakpoint trap (core dumped)

--
It would be great if I could test WHIP / WHEP with Gstreamer standalone (using Rust plugins such as whipsink) but I can't get it to work. I was hoping I could use simple-whip-client but due to the changes in packages, it doesn't seem to be working for me :(

**Comment by mpisat** [comment]
I noticed there is an ubuntu build of this client
https://manpages.ubuntu.com/manpages/jammy/man1/whip-client.1.html

Having a dockerfile using ubuntu latest (?), I'm able to install simple-whip-client and it works
on the other hand it is
bookworm/sid

I don't even know why ubuntu:latest gives bookworm/sid 

FROM ubuntu:latest

ENV TZ=Europe/France
RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
RUN DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC apt-get update && apt-get -y install tzdata

RUN apt-get update && apt-get upgrade -y
RUN apt-get install -y \
    build-essential \
    git curl \
    pkg-config \
    libssl-dev \
    libsrtp2-dev \
    libusrsctp-dev \
    libnice-dev \
    libglib2.0-dev \
    libgstreamer1.0-dev \
    libgstreamer-plugins-base1.0-dev libgstreamer1.0-dev libgstreamer-plugins-base1.0-dev libgstreamer-plugins-bad1.0-dev gstreamer1.0-plugins-base gstreamer1.0-plugins-good gstreamer1.0-plugins-bad gstreamer1.0-plugins-ugly gstreamer1.0-libav gstreamer1.0-tools gstreamer1.0-x gstreamer1.0-alsa gstreamer1.0-gl gstreamer1.0-gtk3 gstreamer1.0-qt5 gstreamer1.0-pulseaudio

**Comment by lminiero** [comment]
As I already explained here: https://github.com/meetecho/simple-whep-client/issues/4#issuecomment-1408690825
probabl7 more recent versions of GStreamer are starting to default to libsoup3 for their own dependencies, while we use libsoup2 in these clients which depend on GStreamer for media aspects. This is probably what causes the conflict. As I explained in that comment, I don't have any immediate plans for changing the code to use libsoup3 instead, with which I'm unfamiliar. PR are welcome though.
